component/org.wso2.carbon.identity.dpop/src/main/java/org/wso2/carbon/identity/dpop/token/binder/DPoPBasedTokenBinder.java

```diff
@@ -241,4 +243,18 @@ private boolean validateDPoPHeader(Object request, TokenBinding tokenBinding) th
         }
         return true;
     }
+
+    public String[] getAllGrantTypes() {
+
+        if (supportedGrantTypes == null) {
+            synchronized (DPoPBasedTokenBinder.class) {
+                if (supportedGrantTypes == null) {
+                    Set<String> allowedGrantSet =
+                            OAuthServerConfiguration.getInstance().getSupportedGrantTypes().keySet();
+                    supportedGrantTypes = new ArrayList<>(allowedGrantSet);
+                }
+            }
+        }
+        return supportedGrantTypes.toArray(new String[supportedGrantTypes.size()]);
```

Comment:
Can't we convert the set to an array straight away?

Reply:
Fixed